rs/xnet/payload_builder/tests/certified_slice_pool.rs

```diff
@@ -131,11 +131,16 @@ proptest! {
             let (prefix, postfix) = unpacked.take_prefix(msg_limit, byte_limit).unwrap();
 
             // Ensure that any limits were respected.
-            if let (Some(msg_limit), Some(prefix)) = (msg_limit, prefix.as_ref()) {
-                assert!(testing::slice_len(prefix) <= msg_limit);
-            }
-            if let (Some(byte_limit), Some(prefix)) = (byte_limit, prefix.as_ref()) {
-                assert!(prefix.count_bytes() <= byte_limit);
+            if let Some(prefix) = prefix.as_ref() {
+                if let Some(msg_limit) = msg_limit {
+                    assert!(testing::slice_len(prefix) <= msg_limit);
+                }
+                if let Some(byte_limit) = byte_limit {
+                    assert!(prefix.count_bytes() <= byte_limit);
+                }
+                if let Some(slice_end) = testing::slice_end(prefix) {
+                    assert!(slice_end <= testing::stream_begin(prefix) + StreamIndex::new(MAX_STREAM_MESSAGES as u64));
+                }
```

Comment:
This is more for the sake of completion since it is never called with number anywhere near large enough to meaningfully check this. Since it is a proptest, I'd rather not extend this to include 10k messages in every single iteration and I'd rather add a small unit test to check for the signals limit.